.github/workflows/build-test-oss.yml

```diff
@@ -0,0 +1,76 @@
+# Main workflow
+name: Build and test changes
+
+on:
+  push:
+    paths-ignore:
+      - 'docs/**'
+      - '*.md'
+      - '*.adoc'
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  linting:
+    name: Run static analysis and linting tools
+    runs-on: ubuntu-20.04
+    timeout-minutes: 15
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 1
+
+      - name: Run linting
+        run: |
+          make lint
+
+  build-test-native:
+    name: Native tests
+    runs-on: ubuntu-20.04
+    timeout-minutes: 30
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 1
+
+      - name: Build OSS image
+        run: |
+          IMAGE_TAG=docker.io/couchbaselabs/observability-stack:${{ github.sha }} tools/build-oss-container.sh
```

Comment:
Nit: should we be using the makefile here for consistency?

Reply:
Yes, I think the makefile doesn't support passing in the value so I need to ensure that is possible